### PR TITLE
APIS-7046: Added new commands and tests

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/apiplatform/modules/commands/applications/domain/models/ApplicationCommand.scala [hunk 1]
```diff
@@ -103,6 +101,14 @@ object ApplicationCommands {
   case class DeclineResponsibleIndividual(code: String, timestamp: Instant)                                                                       extends SubmissionCommand
   case class DeclineResponsibleIndividualDidNotVerify(code: String, timestamp: Instant)                                                           extends SubmissionCommand
   case class DeclineApplicationApprovalRequest(gatekeeperUser: String, reasons: String, timestamp: Instant)                                       extends SubmissionCommand with GatekeeperMixin
+  case class GrantApplicationApprovalRequest(gatekeeperUser: String, timestamp: Instant)                                                          extends SubmissionCommand with GatekeeperMixin
+  case class GrantApplicationApprovalRequestWithWarnings(gatekeeperUser: String, timestamp: Instant, warnings: String, escalatedTo: String)              extends SubmissionCommand
```
Comment (petekirby-ee) [inline src/main/scala/uk/gov/hmrc/apiplatform/modules/commands/applications/domain/models/ApplicationCommand.scala:105]: I think that escalatedTo should be optional

File: src/main/scala/uk/gov/hmrc/apiplatform/modules/commands/applications/domain/models/ApplicationCommand.scala [hunk 2]
```diff
@@ -103,6 +101,14 @@ object ApplicationCommands {
...
+      with GatekeeperMixin
+  case class GrantTermsOfUseApproval(gatekeeperUser: String, timestamp: Instant, reasons: String, escalatedTo: String)                                   extends SubmissionCommand with GatekeeperMixin
```
Comment (petekirby-ee) [inline src/main/scala/uk/gov/hmrc/apiplatform/modules/commands/applications/domain/models/ApplicationCommand.scala:107]: I think that escalatedTo should be optional

**Review by petekirby-ee — CHANGES_REQUESTED** [review]

File: src/main/scala/uk/gov/hmrc/apiplatform/modules/commands/applications/domain/models/ApplicationCommand.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply src/main/scala/uk/gov/hmrc/apiplatform/modules/commands/applications/domain/models/ApplicationCommand.scala:105]: ok will sort

File: src/main/scala/uk/gov/hmrc/apiplatform/modules/commands/applications/domain/models/ApplicationCommand.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply src/main/scala/uk/gov/hmrc/apiplatform/modules/commands/applications/domain/models/ApplicationCommand.scala]: OK will sort

**Review by petekirby-ee — APPROVED** [review]

**Pull request merged**